tests/Application/Commands/DeleteStalePaymentDetailsTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\Application\Commands;
+
+use MatchBot\Application\Commands\DeleteStalePaymentDetails;
+use MatchBot\Tests\Application\DonationTestDataTrait;
+use MatchBot\Tests\Application\StripeFormattingTrait;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Log\NullLogger;
+use Stripe\Service\ChargeService;
+use Stripe\Service\CustomerService;
+use Stripe\Service\PaymentMethodService;
+use Stripe\StripeClient;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Lock\LockFactory;
+
+class DeleteStalePaymentDetailsTest extends TestCase
+{
+    use DonationTestDataTrait;
+    use StripeFormattingTrait;
+
+    public function testDeletingOneStaleCard(): void
```

Comment:
These are long test functions, I think tricky to read through and follow. Is it possible to split them up, maybe just with `// arrange` `// act` `// assert` or `// given` `// when` `// then` comments? Or if each function is testing more than one thing could it be split into multiple test functions?

Reply:
Right I see each test is just testing one thing - there's just quite a bit of code. May be worth extracting some into functions. Can't quite do AAA in the standard order because mocks always switch around the last two A's.